tests/test_otel.py

```diff
@@ -0,0 +1,318 @@
+    "Tracestate": "congo=ucfJifl5GOE,rojo=00f067aa0ba902b7",
+}
+
+
+def decode_id(span, value):
+    if value in ["trace_id", "span_id"]:
+        return binascii.hexlify(getattr(span, value)).decode("utf-8")
+    return value
+
+
+def get_attr(span, name):
+    for value in (a.value for a in span.attributes if a.key == name):
+        return getattr(value, value.WhichOneof("value"))
+
+
+def get_http09(url):
+    def do_get(sock, path):
+        http_send = f"GET {path}\n".encode()
+        sock.sendall(http_send)
+        http_recv = sock.recv(1024)
+        return http_recv.decode("utf-8")
+
+    p = urllib.parse.urlparse(url)
+    with socket.create_connection((p.hostname, p.port)) as sock:
+        recv = do_get(sock, p.path)
+    return recv
+
+
+@pytest.fixture
+def client(nginx):
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+    with niquests.Session(multiplexed=True) as s:
+        yield s
+
+
+def test_response_http09(trace_service, nginx):
+    assert get_http09("http://127.0.0.1:18080/ok") == "OK"
+
+    span = trace_service.get_span()
+
+    assert span.name == "/ok"
+
+
+@pytest.mark.parametrize("http_ver", ["1.1", "2.0", "3.0"])
+@pytest.mark.parametrize(
+    ("path", "status"),
+    [("/ok", 200), ("/err", 500)],
+    ids=["ok", "500 error"],
+)
+def test_response_and_span_attributes(
+    http_ver, path, status, trace_service, client
+):
+    scheme, port = ("http", 18080) if http_ver == "1.1" else ("https", 18443)
+    if http_ver == "3.0":
+        client.quic_cache_layer.add_domain("127.0.0.1", port)
+    r = client.get(f"{scheme}://127.0.0.1:{port}{path}", verify=False)
+
+    span = trace_service.get_span()
+    assert span.name == path
+
+    # Default span attributes
+    assert get_attr(span, "http.method") == "GET"
+    assert get_attr(span, "http.target") == path
+    assert get_attr(span, "http.route") == path
+    assert get_attr(span, "http.scheme") == scheme
+    assert get_attr(span, "http.flavor") == http_ver
+    assert get_attr(span, "http.user_agent") == (
+        f"niquests/{niquests.__version__}"
+    )
+    assert get_attr(span, "http.request_content_length") == 0
+    assert get_attr(span, "http.response_content_length") == len(r.text)
+    assert get_attr(span, "http.status_code") == status
+    assert get_attr(span, "net.host.name") == "localhost"
+    assert get_attr(span, "net.host.port") == port
+    assert get_attr(span, "net.sock.peer.addr") == "127.0.0.1"
+    assert get_attr(span, "net.sock.peer.port") in range(1024, 65536)
+
+
+def test_custom_attributes(trace_service, client):
+    assert client.get("http://127.0.0.1:18080/custom").status_code == 200
+
+    span = trace_service.get_span()
+    assert span.name == "custom_location"
+
+    # Custom span attributes
+    assert get_attr(span, "http.request.completion") == "OK"
+    value = get_attr(span, "http.response.header.content.type")
+    assert value is not None and value.values[0].string_value == "text/plain"
```

Comment:
Just `assert value.values[0].string_value == "text/plain"` should be enough.

Reply:
Fixed.